spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -828,6 +828,26 @@ public void testDefaultSortOrder() {
     assertEquals("Data after compaction should not change", expectedRecords, actualRecords);
   }
 
+  @Test
+  public void testRewriteWithUntranslatedOrUnconvertedFilter() {
```

Comment:
nit: this can potentially be part of existing test `testRewriteDataFilesWithInvalidInputs` but I think it's also fine to leave it here.